### PR TITLE
Graphql cartype

### LINKED ISSUE #16: Add graphql testing for car Type
Add tests for:
- [x] Find one car
- [x] Find all cars
- [x] Find all cars for one user
- [x] Create a car for a user

### PR DESCRIPTION
**Pull request by s-steel**
## Description
Increase testing for graphql car type

## Motivation and Context
Needed tests for the graphql queries

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the README to reflect any changes.

### CODE REVIEW

File: app/graphql/mutations/create_user_car.rb [hunk 1]
```diff
@@ -20,8 +21,17 @@ def resolve(args)
...
+      if car.save
+        {
+          created_car: car,
+          errors: []
+        }
+      else
+        {
+          car: nil,
+          errors: car.errors.full_messages
+        }
```
Comment (ETBassist) [inline app/graphql/mutations/create_user_car.rb:33]: Awesome, thanks for taking care of some of this sad path stuff!!

File: app/graphql/mutations/create_user_car.rb [hunk 2]
```diff
@@ -20,8 +21,17 @@ def resolve(args)
...
-      {car: car}
+      if car.save
+        {
+          created_car: car,
+          errors: []
+        }
+      else
+        {
+          car: nil,
+          errors: car.errors.full_messages
```
Comment (c-bartell) [inline app/graphql/mutations/create_user_car.rb:32]: Great job! We should talk about how we want to handle this in the front end at some point.

File: app/graphql/types/query_type.rb [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+      Car.all
+    end
+
+    field :car, Types::CarType, null: false do
+      argument :id, ID, required: true
+    end
+
+    def car(id:)
+      Car.find(id)
+    end
```
Comment (ETBassist) [inline app/graphql/types/query_type.rb:16]: Awesome work! Should these methods live elsewhere though? Not saying they have to, genuinely just not sure how this should be structured

File: spec/requests/graphql/create_user_car_spec.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      GRAPHQL
+
+      post graphql_path, params: { query: query_string }
+      result = JSON.parse(response.body)
+      expect(result).to have_key('errors')
+      expect(result['errors']).to be_an(Array)
+      expect(result['errors'][0]).to have_key('message')
+      expect(result['errors'][0]['message']).to eq(
+        "Argument 'fuelType' on InputObject 'CreateUserCarInput' is required. Expected type String!"
+      )
```
Comment (c-bartell) [inline spec/requests/graphql/create_user_car_spec.rb:75]: Excellent sad path testing!

**Review by c-bartell — APPROVED** [review]

**Review by ETBassist — APPROVED** [review]
Looks Great! Dropped a question about where methods should be, but more for discussion than for changing. Awesome work!

Great says Grommit!

**Review by Jonathan-M-Wilson — APPROVED** [review]

**Pull request merged**